x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/ccr/action/StatsResponsesTests.java

```diff
@@ -73,18 +69,14 @@ static FollowStatsAction.StatsResponses createStatsResponse() {
         return new FollowStatsAction.StatsResponses(Collections.emptyList(), Collections.emptyList(), responses);
     }
 
-    public void testChunking() throws IOException {
-        final var instance = createTestInstance();
-        int chunkCount = 0;
-        try (var builder = jsonBuilder()) {
-            final var iterator = instance.toXContentChunked(EMPTY_PARAMS);
-            while (iterator.hasNext()) {
-                iterator.next().toXContent(builder, ToXContent.EMPTY_PARAMS);
-                chunkCount += 1;
-            }
-        } // closing the builder verifies that the XContent is well-formed
-
-        var indexCount = instance.getStatsResponses().stream().map(s -> s.status().followerIndex()).distinct().count();
-        assertEquals(instance.getStatsResponses().size() + indexCount * 2 + 2, chunkCount);
+    public void testChunking() {
+        AbstractChunkedSerializingTestCase.assertObjectChunkCount(
+            createTestInstance(),
+            instance -> Math.toIntExact(
+                2 * instance.getStatsResponses().stream().map(s -> s.status().followerIndex()).distinct().count() + instance
+                    .getStatsResponses()
+                    .size() + 2
+            )
+        );
     }
```

Comment:
This and couple of other tests are extending `AbstractChunkedSerializingTestCase`. In such cases test implementation looks identical except the function that calculates the chunk count. Is it worth moving `testChunking` into `AbstractChunkedSerializingTestCase` and adding a new abstract method to provide the chunk count for the test object? This way chunk count test would be enforced in all sub-classes instead of relying on remembering to add one

Reply:
I know what you mean, but `AbstractChunkedSerializingTestCase` (and indeed `AbstractSerializationTestCase`) derivatives are only appropriate for objects that we expect to be able to parse back from their `XContent` representations. That used to apply to many objects because of the HLRC, and there's still a load of legacy in this area, but these days it's less important and indeed almost all of these tests (including the one you link here) are not such tests. So in most cases I expect we still need an explicit test for chunk count.

This whole question probably deserves a bit more of an overhaul than would make sense in this PR.